src/js/node/dgram.ts

```diff
@@ -1,23 +1,1082 @@
-// Hardcoded module "node:dgram"
-// This is a stub! None of this is actually implemented yet.
+// Copyright Joyent, Inc. and other Node contributors.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to permit
+// persons to whom the Software is furnished to do so, subject to the
+// following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN
+// NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR
+// OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE
+// USE OR OTHER DEALINGS IN THE SOFTWARE.
+
 const { hideFromStack, throwNotImplemented } = require("internal/shared");
 
-function createSocket() {
-  throwNotImplemented("node:dgram createSocket", 1630);
+const {
+  FunctionPrototypeBind,
+  ObjectSetPrototypeOf,
+  ReflectApply,
+  SymbolAsyncDispose,
+  SymbolDispose,
+  StringPrototypeTrim,
+  NumberIsNaN,
+} = require("internal/primordials");
+
+const { promisify } = require("node:util");
+const EventEmitter = require("node:events");
+
+const dc = require("node:diagnostics_channel");
```

Comment:
can these be lazier? we only need to load node:diagnostics_channel when the socket is about to be created right? tons of packages will just require a ton of node core modules and then never use them or only use them infrequently-called codepaths, so we should avoid eager initialization as much as possible